TTS_xwing/src/Game/Mechanic/Movement/Maneuver.ttslua

```diff
@@ -0,0 +1,242 @@
+end
+
+function ManeuverStrategy:execute(ship)
+    -- To be overridden by subclasses
+end
+
+function ManeuverStrategy:getBaseSize(ship)
+    local shipData = ship.getTable("Data")
+    local size = shipData.Size or "small"
+    if type(size) == "string" then
+        return igu(Dim.mm_baseSize[size])
+    elseif type(size) == "number" then
+        return igu(size) -- Assuming the size is provided in mm
+    else
+        error("Invalid size type")
+    end
+end
+
+--[[
+    If the forward Vector of the object doesn't line up with the object
+    visually, you can attach an angle value here to correct the forward vector
+    by rotating it around the y-axis by the provided angle in degrees.
+    This will correct an object's forward vector to move it properly.
+    [+ is clockwise, - is counter-clockwise]
+ ]]
+function ManeuverStrategy:getForwardVector(ship)
+    local shipData = ship.getTable("Data")
+    local forwardAdjustment = shipData.forwardAdjustment or 0
+    local forward = ship.getTransformForward():copy()
+    forward = forward:rotateOver("y", forwardAdjustment)
+    return forward:normalized()
+end
+
+-- StraightMove class inherits from MovementStrategy
+local Straight = setmetatable({}, { __index = ManeuverStrategy })
+Straight.__index = Straight
+
+function Straight:new(speed)
+    local instance = ManeuverStrategy:new()
+    instance.speed = igu(XWingTemplates.straight[speed])
+    return setmetatable(instance, self)
+end
+
+function Straight:execute(ship)
+    local forward = self:getForwardVector(ship)
+    local baseSize = self:getBaseSize(ship)
+    local posDelta = (self.speed + baseSize) * forward
+    local newPos = ship.getPosition() + posDelta
+    ship.setPositionSmooth(newPos, false, true)
+end
+
+-- KoiogranTurn class inherits from StraightMove
+local KoiogranTurn = setmetatable({}, { __index = Straight })
+KoiogranTurn.__index = KoiogranTurn
+
+function KoiogranTurn:new(speed)
+    local instance = Straight:new(speed)
+    instance.rotation = 180
+    return setmetatable(instance, self)
+end
+
+function KoiogranTurn:execute(ship)
+    Straight.execute(self, ship)
+    local newRotation = ship.getRotation() + Vector(0, self.rotation, 0)
+    ship.setRotationSmooth(newRotation, false, false)
+end
+
+-- Moves along a curved template
+local ArcMove = setmetatable({}, { __index = ManeuverStrategy })
+ArcMove.__index = ArcMove
+
+function ArcMove:new(radius, angle, bearing, extraRot)
+    local instance = ManeuverStrategy:new()
+    instance.radius = igu(radius)
+    instance.angle = angle
+    instance.bearing = bearing or "right" -- Default to right if no bearing is provided
+    instance.extraRotation = extraRot or 0
+    return setmetatable(instance, self)
+end
+
+function ArcMove:execute(ship)
+    local sign = self.bearing == "right" and 1 or -1 -- + is clockwise, - is counter-clockwise
+    local turnAngle = sign * self.angle
+    local rotAngle = sign * self.extraRotation + turnAngle
```

Comment:
Should there be a paranthesis here?

Reply:
It would improve clarity.

Reply:
Line 110. It doesn't look quite correct without the paranthesis.

Reply:
added more parens